swig/cpp/src/Libyang.cpp

```diff
@@ -0,0 +1,297 @@
+/**
+ * @file Libyang.cpp
+ * @author Mislav Novakovic <mislav.novakovic@sartura.hr>
+ * @brief Implementation of header Libyang.hpp
+ *
+ * @copyright
+ * Copyright 2017 Deutsche Telekom AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+#include <vector>
+
+#include "Xml.hpp"
+#include "Internal.hpp"
+#include "Libyang.hpp"
+#include "Tree_Data.hpp"
+#include "Tree_Schema.hpp"
+
+extern "C" {
+#include "../../../src/libyang.h"
+#include "../../../src/tree_data.h"
+#include "../../../src/tree_schema.h"
+}
+
+using namespace std;
+
+Context::Context(ly_ctx *ctx, S_Deleter deleter) {
```

Comment:
In C++, it's more common to use initialize member variables like this:

```c++
Foo::Foo(Bar *parent, int a, int b):
  parent(parent),
  a(a)
  b(b)
{
}
```

It also shows that there's no name clash, the member variables and arguments can share the same name here.

Reply:
ok.